src/components/decimal/decimal.component.js

```diff
@@ -11,247 +16,261 @@ import LocaleContext from "../../__internal__/i18n-context";
-    const separatorRegex = new RegExp(
-      separator === "." ? `\\${separator}` : separator,
-      "g"
-    );
-    if (
-      valueWithoutNBS.match(errorsWithDelimiter) ||
-      (valueWithoutNBS.match(separatorRegex) || []).length > 1
-    ) {
-      return valueWithoutNBS;
+      const errorsWithDelimiter = new RegExp(
+        `([^A-Za-z0-9]{2,})|(^[^A-Za-z0-9-]+)|([^0-9a-z-,.])|([^0-9-,.]+)|([W,.])$`,
+        "g"
+      );
+      const separator = getSeparator("decimal");
+      const separatorRegex = new RegExp(
+        separator === "." ? `\\${separator}` : separator,
+        "g"
+      );
+      if (
+        valueWithoutNBS.match(errorsWithDelimiter) ||
+        (valueWithoutNBS.match(separatorRegex) || []).length > 1
+      ) {
+        return valueWithoutNBS;
+      }
+
+      const withoutDelimiters = removeDelimiters(valueWithoutNBS);
+      return withoutDelimiters.replace(new RegExp(`\\${separator}`, "g"), ".");
+    },
+    [getSeparator, removeDelimiters]
+  );
+
+  const decimalValue = getSafeValueProp(defaultValue || value || emptyValue);
+  const [stateValue, setStateValue] = useState(
+    isNaN(toStandardDecimal(decimalValue))
+      ? decimalValue
+      : formatValue(decimalValue)
+  );
+
+  const createEvent = useCallback(
+    (formatted, raw) => {
+      return {
+        target: {
+          name,
+          id,
+          value: {
+            formattedValue: formatValue(toStandardDecimal(formatted)),
+            rawValue: raw || toStandardDecimal(formatted),
+          },
+        },
+      };
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+
+  /**
+   * Determine if the component is controlled at the time of call
+   */
+  function isControlled(arg) {
+    return arg !== undefined;
+  }
+
+  const handleOnChange = (ev) => {
+    const val = ev.target.value;
+    setStateValue(val);
+    if (onChange) onChange(createEvent(val));
+  };
+
+  const handleOnBlur = (ev) => {
+    const { value: updatedValue } = ev.target;
+    let event;
+
+    if (updatedValue) {
+      const standardVisible = toStandardDecimal(updatedValue);
+      const formattedValue = isNaN(standardVisible)
+        ? updatedValue
+        : formatValue(standardVisible);
+
+      event = createEvent(formattedValue, standardVisible);
+      setStateValue(formattedValue);
+    } else {
+      event = createEvent(emptyValue);
+      setStateValue(emptyValue);
     }
 
-    const withoutDelimiters = this.removeDelimiters(valueWithoutNBS);
-    return withoutDelimiters.replace(new RegExp(`\\${separator}`, "g"), ".");
+    if (onBlur) onBlur(event);
   };
 
-  getSeparator(separatorType) {
-    const numberWithGroupAndDecimalSeparator = 10000.1;
-    return Intl.NumberFormat(this.props.locale || this.context.locale())
-      .formatToParts(numberWithGroupAndDecimalSeparator)
-      .find((part) => part.type === separatorType).value;
-  }
+  const isComponentControlled = isControlled(value);
```

Comment:
This const is only used in the `useEffect` below so let's just move it into there

Reply:
I had originally used this const elsewhere hence why it is outside of the `useEffect`. I can move it in there easy enough 👍🏻